src/main/java/appeng/menu/implementations/InterfaceTerminalMenu.java

```diff
@@ -196,44 +195,46 @@ public void doAction(final ServerPlayer player, final InventoryAction action, fi
                         }
                     }
                 } else {
-                    ItemHandlerUtil.setStackInSlot(theSlot, 0, playerHand.addItems(theSlot.getStackInSlot(0)));
+                    ItemHandlerUtil.setStackInSlot(theSlot, 0,
+                            ItemHandlerUtil.insert(playerHand, theSlot.getStackInSlot(0), Actionable.MODULATE));
                 }
 
                 break;
             case SPLIT_OR_PLACE_SINGLE:
 
                 if (hasItemInHand) {
-                    ItemStack extra = playerHand.removeItems(1, ItemStack.EMPTY, null);
+                    ItemStack extra = playerHand.extractItem(0, 1, false);
                     if (!extra.isEmpty()) {
-                        extra = interfaceSlot.addItems(extra);
+                        extra = ItemHandlerUtil.insert(theSlot, extra, Actionable.MODULATE);
                     }
                     if (!extra.isEmpty()) {
-                        playerHand.addItems(extra);
+                        ItemHandlerUtil.insert(playerHand, extra, Actionable.MODULATE);
                     }
                 } else if (!is.isEmpty()) {
-                    ItemStack extra = interfaceSlot.removeItems((is.getCount() + 1) / 2, ItemStack.EMPTY, null);
+                    ItemStack extra = theSlot.extractItem(0, (is.getCount() + 1) / 2, false);
                     if (!extra.isEmpty()) {
-                        extra = playerHand.addItems(extra);
+                        extra = ItemHandlerUtil.insert(playerHand, extra, Actionable.MODULATE);
                     }
                     if (!extra.isEmpty()) {
-                        interfaceSlot.addItems(extra);
+                        ItemHandlerUtil.insert(theSlot, extra, Actionable.MODULATE);
                     }
                 }
 
                 break;
             case SHIFT_CLICK:
 
-                final InventoryAdaptor playerInv = InventoryAdaptor.getAdaptor(player);
+                var playerInv = new PlayerMainInvWrapper(player.getInventory());
 
-                ItemHandlerUtil.setStackInSlot(theSlot, 0, playerInv.addItems(theSlot.getStackInSlot(0)));
+                ItemHandlerUtil.setStackInSlot(theSlot, 0,
+                        ItemHandlerUtil.insert(playerInv, theSlot.getStackInSlot(0), Actionable.MODULATE));
 
                 break;
             case MOVE_REGION:
 
-                final InventoryAdaptor playerInvAd = InventoryAdaptor.getAdaptor(player);
+                var playerInvAd = new PlayerMainInvWrapper(player.getInventory());
```

Comment:
Actually the adaptor tried to stack existing items, while `PlayerMainInvWrapper` doesn't... might need to roll a wrapper around it.

Reply:
There might be further issues. E.g. powah will craft stacks of unstackable items which the adaptor or itemhandler for the player hand could not handle and fail in various ways like not being able to craft it in our terminal.

Might not apply here, but should be something we have to verify.